dnf5/context.cpp

```diff
@@ -210,17 +210,27 @@ void Context::print_info(const char * msg) {
 
 
 void Context::update_repo_load_flags_from_specs(const std::vector<std::string> & pkg_specs) {
+    // TODO(jkolarik): unify this behavior with internal libdnf detection of file specs
     // if a pkg_spec contains '/', it's a path and we need to load filelists
     for (auto & spec : pkg_specs) {
         if (spec.find("/") != std::string::npos) {
-            available_repos_load_flags = available_repos_load_flags | libdnf::repo::LoadFlags::FILELISTS;
-            break;
+            add_repo_load_flags(libdnf::repo::LoadFlags::FILELISTS);
+            return;
         }
     }
 }
 
 
-void Context::load_repos(bool load_system, libdnf::repo::LoadFlags flags) {
+void Context::setup_repo_load_flags(const libdnf::repo::RepoWeakPtr & repo) {
+    auto & metadata_types_option = repo->get_config().metadata_types();
+    auto new_flags = flags_to_strings(repo_load_flags);
+    auto configured_flags = metadata_types_option.get_value();
+    new_flags.insert(configured_flags.cbegin(), configured_flags.cend());
+    metadata_types_option.set(libdnf::Option::Priority::RUNTIME, new_flags);
+}
```

Comment:
I like the deduplication, but I am not sure whether this is the right place. If configuration Class for metadata_types will handle it it may be make more sense to me.